### DISCUSSION TITLE
Slider auto focus

### DISCUSSION THREAD

**Discussion by ArturNicolaiewskyPellicioli** [chakra-ui/chakra-ui#5282] [Q&A]
I need set slider with autofocus when the modal renders, already tried with initialFocusRef and with dom manipulation (getElementByName). Does anyone know how to do it?

Project in ReactJs with NextJs.

Note: the same test works with Input, only Slider not worked.

Slider link at doc: https://chakra-ui.com/docs/form/slider

**Comment by malashevskyi** [comment]
Did you try to set initial ref on `SliderThumb`?
```javascript
<Modal initialFocusRef={initialRef}>
 <SliderThumb ref={initialRef} />
...
```

**Reply by ArturNicolaiewskyPellicioli** [reply]
yes, I tried on all slider components :(

**Comment by primos63** [comment]
The `initialRef` used by `initialFocusRef` needs to go on the `SliderThumb` and not the `Slider`. The thumb receives focus.

**Reply by ArturNicolaiewskyPellicioli** [reply]
I tried it. The same thing that works for the input, doesn't work for theSliderThumb or any other slider component.

**Reply by primos63** [reply]
Then you must be doing something wrong.
https://codesandbox.io/s/competent-mayer-7k9pn
